cmd/notary-server/config.go

```diff
@@ -143,21 +143,26 @@ func getTrustService(configuration *viper.Viper, sFactory signerFactory,
 // else parsing will return an error.  A max-age of 0 will disable caching for
 // that type of download (consistent or current).
 func getCacheConfig(configuration *viper.Viper) (utils.CacheControlConfig, utils.CacheControlConfig, error) {
-	var cccs []utils.CacheControlConfig
-	types := []string{"current_metadata", "metadata_by_checksum"}
+	cccs := make(map[string]utils.CacheControlConfig)
+	current, consistent := "current_metadata", "consistent_metadata"
+
+	defaults := map[string]int{
+		current:    int(notary.CurrentMetadataCacheMaxAge.Seconds()),
+		consistent: int(notary.ConsistentMetadataCacheMaxAge.Seconds()),
+	}
+	maxMaxAge := int(notary.CacheMaxAgeLimit.Seconds())
 
-	for _, optionName := range types {
+	var err error
+	for optionName, seconds := range defaults {
 		m := configuration.GetString(fmt.Sprintf("caching.max_age.%s", optionName))
-		if m == "" {
-			continue
-		}
-		seconds, err := strconv.Atoi(m)
-		if err != nil || seconds < 0 || seconds > maxMaxAge {
-			return nil, nil, fmt.Errorf(
-				"must specify a cache-control max-age between 0 and %v", maxMaxAge)
+		if m != "" {
+			seconds, err = strconv.Atoi(m)
+			if err != nil || seconds < 0 || seconds > maxMaxAge {
+				return nil, nil, fmt.Errorf(
+					"must specify a cache-control max-age between 0 and %v", maxMaxAge)
+			}
 		}
-
-		cccs = append(cccs, utils.NewCacheControlConfig(seconds, optionName == "current_metadata"))
+		cccs[optionName] = utils.NewCacheControlConfig(seconds, optionName == current)
 	}
-	return cccs[0], cccs[1], nil
+	return cccs[current], cccs[consistent], nil
```

Comment:
nit: could we add a short sentence to the Godoc for this function about how the current cache config is returned first, followed by the consistent one?

It could even be a short note on the first sentence -- I think it'd be helpful to have a very clear description of the order since it isn't as obvious from the function name itself.